src/plugins/data_source_management/public/components/validation/datasource_form_validation.test.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AuthType } from '../../types';
+import { CreateDataSourceState } from '../create_data_source_wizard/components/create_form/create_data_source_form';
+import { EditDataSourceState } from '../edit_data_source/components/edit_form/edit_data_source_form';
+import {
+  defaultValidation,
+  performDataSourceFormValidation,
+  validateUpdatePassword,
+} from './datasource_form_validation';
+import { mockDataSourceAttributesWithAuth } from '../../mocks';
+
+describe('DataSourceManagement: Form Validation', () => {
+  describe('validate create/edit datasource', () => {
+    let form: CreateDataSourceState | EditDataSourceState = {
+      formErrors: [],
+      formErrorsByField: { ...defaultValidation },
+      title: '',
+      description: '',
+      endpoint: '',
+      auth: {
+        type: AuthType.UsernamePasswordType,
+        credentials: {
+          username: '',
+          password: '',
+        },
+      },
+    };
```

Comment:
We should add some tests for when this structure is not exactly maintained. For example, when `auth` or `credentials` prop is missing.

Reply:
Unit tests are included. Auth & Credentials are always expected to exist on types CreateDataSourceState & EditDataSourceState .. Without those the component will error out and would not even invoke validation methods.

Reply:
We can add this in a separate PR.

Reply:
@mpabba3003 you mentioned "Without those the component will error out and would not even invoke validation methods"... a test should make sure that that happens, which is what I am pointing at.

Reply:
oh ok.. I will write tests for that scenario